File: 기타/숫자 변환하기/code.js
function solution(x, y, n) {
    let count = 1;
    let queue = [x];
    const mySet = new Set();
    if(x===y) return 0;
    while(queue.length > 0){
        let length =queue.length;
        for(let i=0;i<length;i++){
            let x = queue.shift();
            if(x+n===y || x*2===y || x*3===y) return count;
            if(!mySet.has(x+n) && (x+n < y)){
                queue.push(x+n);
                mySet.add(x+n);
            }
            if(!mySet.has(x*2) && (x*2 < y)){
                queue.push(x*2);
                mySet.add(x*2);
            }
            if(!mySet.has(x*3) && (x*3 <y)){
                queue.push(x*3);
                mySet.add(x*3);
            }
        }
        count++;
    }
    return -1;
}

File: algorithm/완전탐색/전력망을 둘로 나누기/code.py
from collections import deque

def solution(n, wires):
    # 인접리스트로 그래프 만들기
    graph = [[] for _ in range(n+1)]
    for a, b in wires:
        graph[a].append(b)
        graph[b].append(a)
        
    # 특정 전선을 끊었을 때 두 전력망의 차이를 구하는 함수
    def get_diff(a, b):
        visited = [False] * (n+1)
        visited[b] = True
        
        queue = deque([a])
        visited[a] = True
        count = 0
        
        while queue:
            cur = queue.popleft()
            count += 1
            
            for next_node in graph[cur]:
                if not visited[next_node]:
                    visited[next_node] = True
                    queue.append(next_node)
                    
        return abs(count - (n - count))
    
    # 모든 전선을 하나씩 끊으며 최솟값 찾기
    min_diff = n
    for a, b in wires:
        min_diff = min(min_diff, get_diff(a, b))
        
    return min_diff
